tlsutil/config.go

```diff
@@ -428,11 +428,16 @@ func (c *Configurator) commonTLSConfig(verifyIncoming bool) *tls.Config {
 	tlsConfig.PreferServerCipherSuites = c.base.PreferServerCipherSuites
 
 	// GetCertificate is used when acting as a server and responding to
-	// client requests. Always return the manually configured cert, because
-	// on the server this is all we have. And on the client, this is the
-	// only sensitive option.
+	// client requests. Default to the manually configured cert, but allow
+	// autoEncrypt cert too so that a client can encrypt incoming
+	// connections without having a manual cert configured.
 	tlsConfig.GetCertificate = func(*tls.ClientHelloInfo) (*tls.Certificate, error) {
-		return c.manual.cert, nil
+		cert := c.manual.cert
+		if cert == nil {
+			cert = c.autoEncrypt.cert
```

Comment:
can `c.autoEncrypt` ever be nil? If it is this will panic.

Reply:
NewConfigurator initializes it: https://github.com/hashicorp/consul/blob/ef5214740712df440ee39f24af3029d2bf52a8ee/tlsutil/config.go#L169-L176.